bin/kafka-run-class.sh

```diff
@@ -20,6 +20,10 @@ then
   exit 1
 fi
 
+# CYGINW == 1 if Cygwin is detected, else 0.
+[[ $(uname -a) =~ "CYGWIN" ]]
+CYGWIN=$((! $?))
```

Comment:
This is just an aspiration, but given that we've had issues with scripts in the past it'd be nice to structure them such that we could turn on `set -e` to more aggressively fail if we encounter issues. I think this test is not safe in that context since it runs the command and then uses `$?` instead of using the command in an if. Could we just turn this into an if or a one-liner to make it safe if we decided to enable `set -e`?